domain-models-runtime/src/main/java/org/folio/rest/impl/AdminAPI.java

```diff
@@ -208,25 +207,19 @@ public void postAdminImportSQL(InputStream entity, Map<String, String> okapiHead
     try {
       String tenantId = TenantTool.tenantId(okapiHeaders);
       String sqlFile = IOUtils.toString(entity, "UTF8");
-      PostgresClient.getInstance(vertxContext.owner(), tenantId).runSQLFile(sqlFile, false, reply -> {
-        if (! reply.succeeded()) {
-          log.error(reply.cause().getMessage(), reply.cause());
-          asyncResultHandler.handle(Future.failedFuture(reply.cause().getMessage()));
-          return;
-        }
-        if (! reply.result().isEmpty()) {
-          // some statements failed, transaction aborted
-          String msg = "postAdminImportSQL failed:\n  " + String.join("\n  ", reply.result());
-          log.error(msg);
-          asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(
-            PostAdminImportSQLResponse.respond400WithTextPlain(msg)));
-          return;
-        }
-        asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(PostAdminImportSQLResponse.respond200(null)));
-      });
+      PostgresClient.getInstance(vertxContext.owner(), tenantId).runSqlFile(sqlFile)
+      .onFailure(e -> {
+        log.error("{}", e.getMessage(), e);
+        asyncResultHandler.handle(Future.succeededFuture(
+            PostAdminImportSQLResponse.respond400WithTextPlain(e.getMessage())));
+      })
+      .onSuccess(x -> asyncResultHandler.handle(Future.succeededFuture(
+          PostAdminImportSQLResponse.respond200(PostAdminImportSQLResponse.headersFor200())))
+      );
     } catch (Exception e) {
-      log.error(e.getMessage(), e);
-      asyncResultHandler.handle(io.vertx.core.Future.failedFuture(e.getMessage()));
+      log.error("{}", e.getMessage(), e);
+      asyncResultHandler.handle(Future.succeededFuture(
+          PostAdminImportSQLResponse.respond500WithTextPlain(e.getMessage())));
```

Comment:
I agree what you have here uses futures the right way. Also, it looks like this isn't a breaking change to modules that rely on this method. Do I have that right?